dataprocspawner/spawner.py

```diff
@@ -1107,6 +1107,19 @@ def _validate_zones(self, region, zones):
 
     return [z for z in zones.split(',') if z in region_zone_letters] or region_zone_letters
 
+  def _get_custom_image_version(self, custom_image):
+    project = custom_image.split('/')[1]
+    image = custom_image.split('/')[4]
+    try:
+      request = self.compute_client.images().get(project=project, image=image)
+      response = request.execute()
+      version = re.split(r'(-debian.*|-ubuntu.*)', response['labels']['goog-dataproc-version'])
+      goog_version = version[0].replace('-', '.') + version[1]
+    except Exception as e: ## pylint: disable=broad-except
+      self.log.info(f'Failed to process Dataproc custom image version:\n\t{e}')
+
+    return goog_version
```

Comment:
Oops, just realized goog_version needs to be initialized to empty string before the try block or else it'll blow up if an exception occurs.

Reply:
sorry, my loss. fixed